pkg/repo/resolver/scripts/Dockerfile.tpl

```diff
@@ -0,0 +1,49 @@
+#  Template Fields
+#  BaseImage   - image to use as base for the build of this image
+#  FromRPMPath - path to the custom RPM directory relative to the resolver image build context in the EIB container
+#  ToRPMPath   - path to the custom RPM directory relative to the resolver image
+#  RegCode     - scc.suse.com registration code
+#  AddRepo     - additional third-party repositories that will be used in the resolution process
+#  CacheDir    - zypper cache directory where all rpm dependencies will be downloaded to
+#  PkgList     - list of packages for which to do the dependency resolution
+FROM {{.BaseImage}}
+
+{{ if and (ne .FromRPMPath "") (ne .ToRPMPath "") -}}
+COPY {{ .FromRPMPath }} {{ .ToRPMPath -}}
+{{ end }}
+
+{{ if ne .RegCode "" -}}
+RUN suseconnect -r {{.RegCode}}
+RUN SLE_SP=$(cat /etc/rpm/macros.sle | awk '/sle/ {print $2};' | cut -c4) && suseconnect -p PackageHub/15.$SLE_SP/x86_64
+RUN zypper ref
+{{ end }}
+
+{{ if ne .AddRepo "" -}}
+RUN counter=1 && \
+    for i in {{.AddRepo}}; \
+    do \
+      zypper ar --no-gpgcheck -f $i addrepo$counter; \
+      counter=$((counter+1)); \
+    done
+{{ end }}
+
+RUN zypper \
+    --pkg-cache-dir {{.CacheDir}} \ 
+    --no-gpg-checks \
```

Comment:
@jdob I recognise the security concerns and I'm with you. We are definitely going to want to expand this to ensure that the additional repo's can be configured accordingly so the resolver ensures that GPG checks are satisfied at least at resolution time. I'm in two minds about whether we would want to also do gpg-checking at installation of the package at first-boot though. In theory, those checks should have been done at resolution time, but the state of the booted system should also be considered "prestine" and packages installed and verified accordingly. Maybe we merge this as-is, and follow it up with an additional PR to resolve the GPG issues as I'm concerned with the size of the PR as it is already.

Reply:
IMO this `--no-gpg-checks` should become dependent on the global `noGPGChecks` property that will be added to the `packages` configuration. Also I am debating whether we need it here at all, since at this step we would have validated the GPGs of each [custom repo](https://github.com/ipetrov117/edge-image-builder/blob/rpm-resolution/pkg/repo/resolver/scripts/Dockerfile.tpl#L25) we have added. Meaning that all RPMs that will be downloaded here should have valid signatures (@jdob, I might be missing something here)? The only reason I see of having it here is that some unsigned RPM could slip in a signed RPM repo (not sure if this is possible).

Reply:
As discussed offline, this will be handled in a follow-up PR directly after this one.